docs/proposals/promote-dependencies-automatically/README.md

```diff
@@ -0,0 +1,196 @@
+
+#### Failure Resolution
+
+When promoting a resource, if the resource to be promoted and its dependent resources fail to be promoted to the Karmada control plane(validation failed or other), the promotion operation will fail and will not promote any resources to the Karmada control plane.
+
+### Example
+
+Assuming we deploy a workload CR named `myapp` and its referenced ConfigMap called `my-config` in the member cluster member1.
+
+```yaml
+apiVersion: apps/v1
+kind: Workload
+metadata:
+  name: myapp
+  labels:
+    app: myapp
+spec:
+  replicas: 1
+  selector:
+    matchLabels:
+      app: myapp
+  template:
+    metadata:
+      labels:
+        app: myapp
+    spec:
+      containers:
+        - image: nginx
+          name: nginx
+          volumeMounts:
+            - name: configmap
+              mountPath: "/configmap"
+      volumes:
+        - name: configmap
+          configMap:
+            name: my-config
+---
+apiVersion: v1
+kind: ConfigMap
+metadata:
+  name: my-config
+data:
+  nginx.properties: |
+    proxy-connect-timeout: "10s"
+    proxy-read-timeout: "10s"
+    client-max-body-size: "2m"
+```
+After executing the `karmadactl promote workload myapp -c member1 -d=true` command, we can obtain the Resource Binding and Work object corresponding to the workload `myapp` and ConfigMap `my-config` in the Karmada control plane. This indicates that both `myapp` and its dependent resource `my-config` have been successfully promoted to the Karmada control plane.
+
+When deleting `myapp` from the Karmada control plane, `myapp` in the member cluster member1 will also be deleted.
+
+### Test Plan
+
+#### UT
+
+Testing coverage for new methods.
+
+#### E2E
+
+| Test Case                                                 | Case Description                                                                                                                |
+|-----------------------------------------------------------|---------------------------------------------------------------------------------------------------------------------------------|
+| Promote dependencies of native resource automatically     | Test if dependencies of native resource automatically promote to karmada.                                                       |
+| Promote dependencies of thirdparty resource automatically | Test if dependencies of thirdparty resource automatically promote to karmada.                                                   |
+| Promote dependencies of custom resource automatically     | Test if dependencies of custom resource automatically promote to karmada(use lua configuration way and webhook interpret way).  |
+
+
+## Alternatives
+### TODO: Port forward the request to webhook interpret service in the cluster
```

Comment:
Do we need to use `port forward` to request webhook? @jwcesign 
If yes, this chapter should not be presented in `Alternatives`.

Reply:
We still need it. As the comments: https://github.com/karmada-io/karmada/pull/4100#discussion_r1361744410